src/Diagnostics/atmos_mass_energy_loss.jl

```diff
@@ -0,0 +1,120 @@
+# AtmosMassEnergyLoss
+#
+# Dump mass and energy loss
+
+using ..Atmos
+using ..BalanceLaws
+using ..Mesh.Topologies
+using ..Mesh.Grids
+using ..MPIStateArrays
+
+"""
+    setup_atmos_mass_energy_loss(
+        ::ClimateMachineConfigType,
+        interval::String,
+        out_prefix::String,
+        writer = NetCDFWriter(),
+        interpol = nothing,
+    )
+
+Create and return a `DiagnosticsGroup` containing the
+"AtmosMassEnergyLoss" diagnostics for Atmos LES and GCM
+configurations. All the diagnostics in the group will run at the
+specified `interval`, be interpolated to the specified boundaries
+and resolution, and written to files prefixed by `out_prefix`
+using `writer`.
+"""
+function setup_atmos_mass_energy_loss(
+    ::ClimateMachineConfigType,
+    interval::String,
+    out_prefix::String,
+    writer = NetCDFWriter(),
+    interpol = nothing,
+)
+    @assert isnothing(interpol)
+
+    return DiagnosticsGroup(
+        "AtmosMassEnergyLoss",
+        Diagnostics.atmos_mass_energy_loss_init,
+        Diagnostics.atmos_mass_energy_loss_fini,
+        Diagnostics.atmos_mass_energy_loss_collect,
+        interval,
+        out_prefix,
+        writer,
+        interpol,
+    )
+end
+
+# store initial mass and energy
+Base.@kwdef mutable struct AtmosMassEnergyLoss
+    Σρ₀::Float64 = 0.0
+    Σρe₀::Float64 = 0.0
+end
+const AtmosMassEnergyLoss₀ = AtmosMassEnergyLoss()
+
+function atmos_mass_energy_loss_init(dgngrp, currtime)
+    mpicomm = Settings.mpicomm
+    mpirank = MPI.Comm_rank(mpicomm)
+    dg = Settings.dg
+    bl = dg.balance_law
+    Q = Settings.Q
+    FT = eltype(Q)
+
+    ρ_idx = varsindices(vars_state(bl, Prognostic(), FT), "ρ")
+    ρe_idx = varsindices(vars_state(bl, Prognostic(), FT), "ρe")
```

Comment:
```suggestion
    ρ_idx = varsindices(vars_state(bl, Prognostic(), FT), :ρ)
    ρe_idx = varsindices(vars_state(bl, Prognostic(), FT), :ρe)
```
I think this works with strings or symbols, but should we lean towards symbols since they can be used on the GPU?

Reply:
Wasn't sure if you could do something like `:moisture.ρq_tot`, but it turns out you can (with `:(moisture.ρq_tot)`). I can change it, but does it matter @jkozdon?

Reply:
`varsindices` doesn't work on the GPU so I think this a question of preference. I added the string version specifically because I didn't like `:(moisture.ρq_tot)`.

Reply:
Ah, I was thinking of `varsindex`. I think the syntax that works is `varsindex(..., :moisture, :ρq_tot))` for `varsindex`. I think the `varsindices` are for when we want multiple variable indexes.

Reply:
> Wasn't sure if you could do something like `:moisture.ρq_tot`, but it turns out you can (with `:(moisture.ρq_tot)`). I can change it, but does it matter @jkozdon?

Fine with me either way. I'd say symbols or strings is all personal preference.

Reply:
Pretty much. Another difference is that`varsindex` returns a range whereas `varsindices` returns a tuple of integers. I don't think that `varsindex` with multiple `Symbol` arguments works on the GPU either but I could be wrong. Haven't tried it lately.

Reply:
Here's an example we've used on the ocean side that works

https://github.com/CliMA/ClimateMachine.jl/blob/d5df1495500ae78a7fca79ddf88ca5b5561f12f3/src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl#L690-L699

If your symbol returns a range, you can just typecast it to a `tuple()` and then it will work on the GPU as well. Haven't tried to pass multiple symbols as arguments to `varsindex` tho.

Reply:
@mwarusz meant inside a device kernel (not in the host-side code that generates device code).

Reply:
I don't think specifying `(:moisture, :ρq_tot)` is an improvement over `"moisture.ρq_tot"`. Also, this code is not expected to run on the GPU anyway. The `weightedsum` might, but that's internal to `MPIStateArrays` and independent of how we find which index to use.